### PR TITLE
#6193: IHtmlFilter and TokenFilter improvements and bugfixes

### LINKED ISSUE #6193: ContentHandler TokenFilter and Element TokenFilter can't cohabit
**Repro** on dev branch
- In an html widget, use a token that need Content data, e.g #{ Content.Id }.
- So, enable the `Tokens Html Filter` feature (will be used by the body part).
- In an Layout element (text, html...) use such a token, e.g the same as above.
- So, enable the `Element Tokens` feature used by the html, text... elements.

**Result**
- The token in the html widget is not evaluated correctly.
- Note: may depends on the token filters ordering (see below).

**Analyse**

The 1st TokensFilter inherit from ContentHandler, not the 2nd one (IElementFilter), context data are not passed in the same way and can be empty. So, a first running filter can kill the token.

Maybe the 1st filter (that is an html filter) should not be coupled to a content type by exposing a ProcessContent() method with a data context parameter... More details in my last comment here #6155.

**Fix**

To allow such filters to cohabit, in Tokenizer.cs, I've found a fix by testing if the replacement value is null and then do nothing on the token. I will do a PR.

Best

### PR DESCRIPTION
**Pull request by sfmskywalker**
This changeset unifies the two separate TokenFilter implementations (one for BodyPart, TextField, etc and another one for certain elements such as Html).
It also fixes a bug with the TokenFilter when executing for HtmlWidget, where the wrong content item is being recorded by the handler (the original implementation).
Thirdly, it removes awkward code repetition by moving filter execution into a dedicated HtmlFilterRunner service.

Fixes #6193

### CODE REVIEW

**Comment by sfmskywalker** [comment]
This also includes @jtkech 's work on fixing an encoding issue and execution order of the TokenFilter.

File: src/Orchard.Web/Modules/Orchard.Tags/Feeds/TagFeedQuery.cs [hunk 1]
```diff
@@ -1,34 +1,27 @@
...
-using Orchard.Utility.Extensions;
 using Orchard.Core.Feeds;
 using Orchard.Tags.Services;
 using Orchard.Localization;
 using System.Web.Routing;
 using Orchard.Environment.Extensions;
 
-namespace Orchard.Tags.Feeds {
+namespace Orchard.Tags.Feeds
+{
```
Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.Tags/Feeds/TagFeedQuery.cs]: What is that ?

File: src/Orchard/Services/IHtmlFilterRunner.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿using System.Collections.Generic;
+using Orchard.ContentManagement;
+
+namespace Orchard.Services {
+    public interface IHtmlFilterRunner : IDependency {
```
Comment (sebastienros) [inline src/Orchard/Services/IHtmlFilterRunner.cs]: I don't like the name `Runner`. We don't have such things right now. We have `Processor`, `Coordinator`, probably other choices but I don't like `runner`.

**Comment by sebastienros** [comment]
Minus the comments it looks good.

**Comment by jtkech** [comment]
Does this one only waiting for a name and a curly brace rule correction?
And maybe better to target 1.10.x?

Best.

**Review by BenedekFarkas — APPROVED** [review]

File: src/Orchard.Web/Modules/Orchard.Tags/Feeds/TagFeedQuery.cs (hunk 1 above)
Comment (BenedekFarkas) [reply src/Orchard.Web/Modules/Orchard.Tags/Feeds/TagFeedQuery.cs]: Fixed in d95f81259e96f5e23c6c263cb49d9913fb45b5cf

**Comment by BenedekFarkas** [comment]
@sebastienros tested, good to go!

**Pull request merged**